lib/Conversion/CombToAIG/CombToAIG.cpp

```diff
@@ -668,6 +668,23 @@ struct CombSubOpConversion : OpConversionPattern<SubOp> {
   }
 };
 
+// Construct a full adder for three 1-bit inputs.
+std::pair<Value, Value> fullAdder(ConversionPatternRewriter &rewriter,
+                                  Location loc, Value a, Value b, Value c) {
+
+  Value sum =
+      rewriter.createOrFold<comb::XorOp>(loc, ArrayRef<Value>{a, b, c}, true);
+
+  auto carry = rewriter.createOrFold<comb::OrOp>(
+      loc,
+      ArrayRef<Value>{rewriter.createOrFold<comb::AndOp>(loc, a, b),
+                      rewriter.createOrFold<comb::AndOp>(loc, b, c),
+                      rewriter.createOrFold<comb::AndOp>(loc, c, a)},
+      true);
+
+  return {sum, carry};
```

Comment:
I've seen alternate forms that use an intermediate XOR result to get rid of one of the AND gates:
```
tmp = a ^ b
sum = tmp ^ c
carry = (a & b) | (tmp & c)
```
I don't know if this actually simplifies anything in practice. Maybe it's useful to have only 2-input gates? No idea, just a random thought 😃